In [2]:
import h2o
h2o.init()

Checking whether there is an H2O instance running at http://localhost:54321 ..... not found.
Attempting to start a local H2O server...
; Java HotSpot(TM) 64-Bit Server VM (build 25.211-b12, mixed mode)
  Starting server from C:\Users\malbalawi\AppData\Roaming\Python\Python37\site-packages\h2o\backend\bin\h2o.jar
  Ice root: C:\Users\malbalawi\AppData\Local\Temp\10\tmpayqxrqs_
  JVM stdout: C:\Users\malbalawi\AppData\Local\Temp\10\tmpayqxrqs_\h2o_malbalawi_started_from_python.out
  JVM stderr: C:\Users\malbalawi\AppData\Local\Temp\10\tmpayqxrqs_\h2o_malbalawi_started_from_python.err
  Server is running at http://127.0.0.1:54321
Connecting to H2O server at http://127.0.0.1:54321 ... successful.


--------------------------  ------------------------------------------
H2O cluster uptime:         06 secs
H2O cluster timezone:       America/Chicago
H2O data parsing timezone:  UTC
H2O cluster version:        3.24.0.4
H2O cluster version age:    20 days
H2O cluster name:           H2O_from_python_malbalawi_gx3o5a
H2O cluster total nodes:    1
H2O cluster free memory:    1.653 Gb
H2O cluster total cores:    0
H2O cluster allowed cores:  0
H2O cluster status:         accepting new members, healthy
H2O connection url:         http://127.0.0.1:54321
H2O connection proxy:
H2O internal security:      False
H2O API Extensions:         Amazon S3, Algos, AutoML, Core V3, Core V4
Python version:             3.7.3 final
--------------------------  ------------------------------------------

In [3]:
url="http://h2o-public-test-data.s3.amazonaws.com/smalldata/iris/iris_wheader.csv"
iris= h2o.import_file(url)

Parse progress: |█████████████████████████████████████████████████████████| 100%


In [4]:
iris

sepal_len sepal_wid petal_len petal_wid class 
 
 
 5.1 3.5 1.4 0.2 Iris-setosa 
 4.9 3 1.4 0.2 Iris-setosa 
 4.7 3.2 1.3 0.2 Iris-setosa 
 4.6 3.1 1.5 0.2 Iris-setosa 
 5 3.6 1.4 0.2 Iris-setosa 
 5.4 3.9 1.7 0.4 Iris-setosa 
 4.6 3.4 1.4 0.3 Iris-setosa 
 5 3.4 1.5 0.2 Iris-setosa 
 4.4 2.9 1.4 0.2 Iris-setosa 
 4.9 3.1 1.5 0.1 Iris-setosa

In [11]:
train , test= iris.split_frame([0.8])

In [12]:
train.summary()

sepal_len sepal_wid petal_len petal_wid class 
 
 
 type real real real real enum 
 mins 4.4 2.0 1.2 0.1 
 mean 5.8883333333333345 3.0533333333333323 3.835833333333332 1.2358333333333331 
 maxs 7.9 4.4 6.9 2.5 
 sigma 0.8527940566638775 0.43481539274766706 1.7792830126811439 0.7681469389953689 
 zeros 0 0 0 0 
 missing 0 0 0 0 0 
 0 5.1 3.5 1.4 0.2 Iris-setosa 
 1 4.9 3.0 1.4 0.2 Iris-setosa 
 2 4.7 3.2 1.3 0.2 Iris-setosa 
 3 5.0 3.6 1.4 0.2 Iris-setosa 
 4 5.4 3.9 1.7 0.4 Iris-setosa 
 5 4.6 3.4 1.4 0.3 Iris-setosa 
 6 5.0 3.4 1.5 0.2 Iris-setosa 
 7 4.4 2.9 1.4 0.2 Iris-setosa 
 8 4.9 3.1 1.5 0.1 Iris-setosa 
 9 5.4 3.7 1.5 0.2 Iris-setosa

In [13]:
train.nrows

120

In [19]:
test.nrows

30

In [14]:
from h2o.estimators.random_forest import H2ORandomForestEstimator

# Create an instance of it called mRF

In [15]:
mRF= H2ORandomForestEstimator()
mRF.train(["sepal_len","sepal_wid","petal_len","petal_wid"], "class",train)

drf Model Build progress: |███████████████████████████████████████████████| 100%


In [16]:
mRF

Model Details
H2ORandomForestEstimator :  Distributed Random Forest
Model Key:  DRF_model_python_1560877108989_2


ModelMetricsMultinomial: drf
** Reported on train data. **

MSE: 0.035528448002287424
RMSE: 0.18848991485564268
LogLoss: 0.11138642710092769
Mean Per-Class Error: 0.049571603427172574
Confusion Matrix: Row labels: Actual class; Column labels: Predicted class



Iris-setosa    Iris-versicolor    Iris-virginica    Error      Rate
-------------  -----------------  ----------------  ---------  -------
39             0                  0                 0          0 / 39
0              35                 3                 0.0789474  3 / 38
0              3                  40                0.0697674  3 / 43
39             38                 43                0.05       6 / 120

Top-3 Hit Ratios: 


k    hit_ratio
---  -----------
1    0.95
2    1
3    1

Scoring History: 


timestamp            duration    number_of_trees    training_rmse        training_logloss     training_classification_error
---  -------------------  ----------  -----------------  -------------------  -------------------  -------------------------------
     2019-06-18 13:11:05  0.027 sec   0.0                nan                  nan                  nan
     2019-06-18 13:11:05  0.071 sec   1.0                0.35064343665913833  3.4654351057839965   0.13114754098360656
     2019-06-18 13:11:05  0.097 sec   2.0                0.30684824287560153  2.3215884099064614   0.09090909090909091
     2019-06-18 13:11:05  0.125 sec   3.0                0.30565967194176336  2.550870054067121    0.09278350515463918
     2019-06-18 13:11:05  0.131 sec   4.0                0.280814428871987    2.2803672730709468   0.07407407407407407
---  ---                  ---         ---                ---                  ---                  ---
     2019-06-18 13:11:05  0.515 sec   46.0               0.18734789621640519  0.10971011727598433  0.05
     2019-06-18 13:11:05  0.523 sec   47.0               0.18820354877671694  0.11053608629692076  0.05
     2019-06-18 13:11:05  0.528 sec   48.0               0.18914234628608512  0.11133992085125753  0.05
     2019-06-18 13:11:05  0.536 sec   49.0               0.1878574812835787   0.11058644714792391  0.05
     2019-06-18 13:11:05  0.542 sec   50.0               0.18848991485564268  0.11138642710092769  0.05


See the whole table with table.as_data_frame()
Variable Importances: 


variable    relative_importance    scaled_importance    percentage
----------  ---------------------  -------------------  ------------
petal_wid   1819.89                1                    0.529702
petal_len   1220.19                0.670476             0.355152
sepal_len   321.294                0.176546             0.0935167
sepal_wid   74.3117                0.0408331            0.0216294

 # 6 Erorrs in the traing date out of 120 samples which is %95

In [17]:
p=mRF.predict(test)

drf prediction progress: |████████████████████████████████████████████████| 100%


In [18]:
mRF.model_performance(test)


ModelMetricsMultinomial: drf
** Reported on test data. **

MSE: 0.028936792014681523
RMSE: 0.1701081773892176
LogLoss: 0.09619550592049199
Mean Per-Class Error: 0.047619047619047616
Confusion Matrix: Row labels: Actual class; Column labels: Predicted class



Iris-setosa    Iris-versicolor    Iris-virginica    Error      Rate
-------------  -----------------  ----------------  ---------  ------
11             0                  0                 0          0 / 11
0              12                 0                 0          0 / 12
0              1                  6                 0.142857   1 / 7
11             13                 6                 0.0333333  1 / 30

Top-3 Hit Ratios: 


k    hit_ratio
---  -----------
1    0.966667
2    1
3    1

# One Erorr in the test date out of 30 samples which is %96 